### ISSUE TITLE
NAPALM connection plugin documentation

### ISSUE THREAD

**Issue by ryanmerolle** [ansible-collections/ansible.netcommon#50]
##### SUMMARY
Given the  connection plugin for NAPALM as of Ansible 2.8, it would help if some clarity around how to use said connection plugin was documented.

For example, if the connection plugin was leveraged, what task would be used?  I assume the napalm specific tasks, but how would a be structured given NAPALM is now being used as the connection method directly and leveraging the Ansible credentials.

##### ISSUE TYPE
- Documentation Report

##### COMPONENT NAME
napalm connection plugin
napalm.py

##### ANSIBLE VERSION
2.9.9

**Comment by Qalthos** [comment]
Yes, the documentation around the napalm connection plugin is shaky. This is mostly because it was never actually used for anything.

For some background information, the plugin was made to see if there was any interest in having a NAPALM backend for ansible playbooks. While those playbooks would not be able to use existing ansible modules, however modules like https://github.com/napalm-automation/napalm-ansible/blob/develop/napalm_ansible/modules/napalm_cli.py could be vastly simplified into something like https://gist.github.com/Qalthos/05af40490808fc3c4eef3b5952d1cdcb.

And then I proceeded to hear absolutely no one asking about NAPALM support ever again until this very issue. I saw mild disinterest in using this connection from the people originally interested in the plugin's development, and the plugin was scheduled to be deprecated from ansible.netcommon as of 1.0.0 as I had no evidence, circumstantial or otherwise, that anyone had ever even tried to use the plugin.

If you are still interested in the plugin, that's great! The plugin isn't yet deprecated, and the plan was always to rescind the deprecation if anyone ever showed interest. Ideally, if there is any uptake of the plugin, it would eventually graduate out of ansible.netcommon and into either community.network or into some napalm-specific collection where the people actually using it would have greater control over the plugin. Otherwise, I go back to the deprecation PR and wait for someone else to ask me about it.
